[All hosts](NAA SSO) Updates for NAA GA and WXP online support (#5410)
OfficeDev/office-js-docs-pr: docs/develop/enable-nested-app-authentication-in-your-add-in.md

## NAA supported accounts and hosts

NAA supports both Microsoft Accounts and Microsoft Entra ID (work/school) identities. It doesn't support [Azure Active Directory B2C](/azure/active-directory-b2c/overview) for business-to-consumer identity management scenarios. For more information on NAA requirements, see [Nested app auth requirement set](/javascript/api/requirement-sets/common/nested-app-auth-requirement-sets).

## Register your single-page application

> 

Reviewer: @davidchesnut  Do we need to add a note/line about needing another redirect to your actual domain (like a normal SPA app) for when on the web in WXP, given brk-multihub only covers the host applications and OWA which do brokering?

Author: I added a note about this later as well as steps in the app registration process.

Trusted broker groups are dynamic by design and can be updated in the future to include additional hosts where your add-in may use NAA flows. Currently the brk-multihub group includes Office Word, Excel, PowerPoint, Outlook, and Teams (for when Office is activated inside).

> [!IMPORTANT]

> > For Word, Excel, and PowerPoint in the browser, you also need an additional redirect since the browser uses a standard authentication flow. The SPA redirect URI must reference the HTML page where you will use the MSAL.js library to request tokens through NAA.

Author: No. They use the same code pattern.

    - Set **Name** to `contoso-office-add-in-sso`.
    - Set **Supported account types** to **Accounts in any organizational directory (any Azure AD directory - multitenant) and personal Microsoft accounts (e.g. Skype, Xbox)**.
    - Set **Redirect URI** to use the platform **Single-page application (SPA)** and the URI to `brk-multihub://localhost:3000`. This redirect assumes you are testing your add-in from a localhost server.

    :::image type="content" source="../images/azure-portal-create-application-registration-nested-app-authentication.png" alt-text="Register an application pane with name and supported account completed.":::

1. Select **Register**. A message is displayed stating that the application registration was created.

    :::image type="content" source="../images/azure-portal-application-created-message.png" alt-text="Message stating that the application registration was created.":::

1. Copy and save the value for the **Application (client) ID**. You'll use it in a later procedure.

    :::image type="content" source="../images/azure-portal-copy-client-id.png" alt-text="App registration pane for Contoso displaying the client ID and directory ID.":::

> If your add-in supports Word, Excel, or PowerPoint in a browser, you must add an SPA redirect URI for your task pane page. Use the following steps to add an SPA redirect URI for your task pane page.

Reviewer 2: ```suggestion
If your add-in supports Word, Excel, or PowerPoint on the web, you must add an SPA redirect URI for your task pane page. Use the following steps to add an SPA redirect URI for your task pane page.
```
